x/market/keeper/deal.go

```diff
@@ -198,3 +198,19 @@ func (k Keeper) SetCurNumData(deal types.Deal) {
 	curNumData := deal.GetCurNumData() + 1
 	deal.CurNumData = curNumData
 }
+
+func (k Keeper) Verify(ctx sdk.Context, validatorAddr sdk.AccAddress, unSignedCert types.UnsignedDataValidationCertificate, cert types.DataValidationCertificate) (bool, error) {
+	validatorAcc := k.accountKeeper.GetAccount(ctx, validatorAddr)
+
+	unSignedMarshaled, err := unSignedCert.Marshal()
+	if err != nil {
+		return false, sdkerrors.Wrapf(err, "invalid marshaled value")
+	}
+
+	isValid := validatorAcc.GetPubKey().VerifySignature(unSignedMarshaled, cert.GetSignature())
+	if !isValid {
+		return false, sdkerrors.Wrapf(types.ErrInvalidSignature, "%s", cert.GetSignature())
+	}
+
+	return isValid, nil
+}
```

Comment:
```suggestion
func (k Keeper) Verify(ctx sdk.Context, validatorAddr sdk.AccAddress, unSignedCert types.UnsignedDataValidationCertificate, cert types.DataValidationCertificate) error {
	validatorAcc := k.accountKeeper.GetAccount(ctx, validatorAddr)

	unSignedMarshaled, err := unSignedCert.Marshal()
	if err != nil {
		return sdkerrors.Wrapf(err, "invalid marshaled value")
	}

	isValid := validatorAcc.GetPubKey().VerifySignature(unSignedMarshaled, cert.GetSignature())
	if !isValid {
		return sdkerrors.Wrapf(types.ErrInvalidSignature, "%s", cert.GetSignature())
	}

	return nil
}
```
I think the `bool` value and `error` play a similar role.
How about removing the bool?